controllers/drplacementcontrol.go

```diff
@@ -0,0 +1,1372 @@
+	}
+
+	if deployed && d.isUserPlRuleUpdated(homeCluster) {
+		// If for whatever reason, the DRPC status is missing (i.e. DRPC could have been deleted mistakingly and
+		// recreated again), we should update it with whatever status we are at.
+		if d.getLastDRState() == rmn.DRState("") {
+			d.instance.Status.PreferredDecision = d.userPlacementRule.Status.Decisions[0]
+			d.setDRState(rmn.Deployed)
+		}
+
+		return done, nil
+	}
+
+	return d.startDeploying(homeCluster, homeClusterNamespace)
+}
+
+func (d *DRPCInstance) getHomeCluster() (string, string) {
+	// Check if the user wants to use the preferredCluster
+	homeCluster := ""
+	homeClusterNamespace := ""
+
+	if d.instance.Spec.PreferredCluster != "" {
+		homeCluster = d.instance.Spec.PreferredCluster
+		homeClusterNamespace = homeCluster
+	}
+
+	if homeCluster == "" && d.drpcPlacementRule != nil && len(d.drpcPlacementRule.Status.Decisions) != 0 {
+		homeCluster = d.drpcPlacementRule.Status.Decisions[0].ClusterName
+		homeClusterNamespace = d.drpcPlacementRule.Status.Decisions[0].ClusterNamespace
+	}
+
+	return homeCluster, homeClusterNamespace
+}
+
+// isDeployed check to see if the initial deployment is already complete to this
+// homeCluster or elsewhere
+func (d *DRPCInstance) isDeployed(homeCluster string) (string, bool) {
+	if d.isVRGDeployedAndProtected(homeCluster) {
+		d.log.Info(fmt.Sprintf("Already deployed to %s. Last state: %s",
+			homeCluster, d.getLastDRState()))
+
+		return homeCluster, true
+	}
+
+	clusterName, found := d.isAlreadyDeployedElsewhere(homeCluster)
+	if found {
+		errMsg := fmt.Sprintf("Failed to place deployment on cluster %s, as it is active on another cluster",
+			homeCluster)
+		d.log.Info(errMsg)
+
+		return clusterName, true
+	}
+
+	return "", false
+}
+
+func (d *DRPCInstance) isUserPlRuleUpdated(homeCluster string) bool {
+	return len(d.userPlacementRule.Status.Decisions) > 0 &&
+		d.userPlacementRule.Status.Decisions[0].ClusterName == homeCluster
+}
+
+// isAlreadyFullyDeployedAndProtected will check whether a VRG exists in the targetCluster and
+// whether it is in protected state, and primary.
+func (d *DRPCInstance) isVRGDeployedAndProtected(targetCluster string) bool {
+	d.log.Info(fmt.Sprintf("isAlreadyDeployedAndProtected? - %+v", d.vrgs))
+
+	vrg := d.getCachedVRG(targetCluster)
+	if vrg == nil {
+		return false
+	}
+
+	if !d.isVRGProtected(vrg) {
+		return false
+	}
+
+	return d.isVRGPrimary(vrg)
+}
+
+func (d *DRPCInstance) getCachedVRG(clusterName string) *rmn.VolumeReplicationGroup {
+	vrg, found := d.vrgs[clusterName]
+	if !found {
+		d.log.Info("VRG not found on cluster", "clusterName", clusterName)
+
+		return nil
+	}
+
+	return vrg
+}
+
+func (d *DRPCInstance) isVRGProtected(vrg *rmn.VolumeReplicationGroup) bool {
+	// TODO:
+	// 		UNCOMMENT THIS CODE ONCE PR #255 IS MERGED
+	//
+	// dataProtectedCondition := findCondition(vrg.Status.Conditions, VRGConditionTypeClusterDataProtected)
+	// if dataProtectedCondition == nil {
```

Comment:
Should we also check the dataProtectedCondition.Status == True?

Reply:
That will be reworked.  I was waiting for Veera's PR to finalize.